src/ffi.md

```diff
@@ -248,6 +248,72 @@ the release of these resources (especially in the case of panic).
 
 For more about destructors, see the [Drop trait](../std/ops/trait.Drop.html).
 
+## Calling Rust code from C
+
+You may wish to compile Rust code in a way so that it can be called from C.
+This is fairly easy, but requires a few things.
+
+### Rust side
+
+First, we assume you have a lib crate named as `rust_from_c`.
+`lib.rs` should have Rust code as following:
+
+```rust
+#[no_mangle]
+pub extern "C" fn hello_from_rust() {
+    println!("Hello from Rust!");
+}
+# fn main() {}
+```
+
+The `extern "C"` makes this function adhere to the C calling convention, as discussed above in "[Foreign Calling Conventions]".
+The `no_mangle` attribute turns off Rust's name mangling, so that it is easier to link to.
```

Comment:
```suggestion
The `no_mangle` attribute turns off Rust's name mangling, so that it has a well defined symbol to link to.
```

I realize you just moved this from below, but the way it is written seems strange to me.  With a mangled name, I don't think you'd be able to guess the symbol name at all, which makes linking mostly impossible?

Reply:
I moved it because we have the "Callbacks from C code to Rust functions" section below. Applied your suggestion, is it okay with you now?